src/domain/occurrence/eventOccurrenceForm/ValidationSchema.ts

```diff
@@ -95,3 +98,38 @@ export default Yup.object().shape({
     ),
   oneGroupFills: Yup.boolean(),
 });
+
+export const testDateWithEnrolmentValidationSchema = (
+  dateField: Yup.DateSchema,
+  pEvent: PEventFieldsFragment | undefined,
+  schema: Yup.ObjectSchema<object>
+) => {
+  const { enrolmentStart, enrolmentEndDays } = pEvent ?? {};
+
+  if (!enrolmentStart || !enrolmentEndDays) {
+    return schema;
+  }
+
+  function testEnrolmentEndDays(
+    enrolmentStart: Date,
+    enrolmentEndDays: number,
+    schema: Yup.DateSchema<Date>
+  ) {
+    const minDate =
+      enrolmentEndDays > 0
+        ? addDays(enrolmentStart, enrolmentEndDays)
+        : enrolmentStart;
+    minDate.setHours(0, 0, 0, 0);
+    return schema.min(minDate, () => ({
+      key: VALIDATION_MESSAGE_KEYS.DATE_MIN,
+      min: formatDate(minDate),
+    }));
+  }
```

Comment:
I would move this function one outside of this function because all parameters are passed anyway. To that would be easier to read

Reply:
Moved.